tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java

```diff
@@ -0,0 +1,259 @@
+    repo.init();
+    repo.setFakeUsersData();
+    fileUtils.writeStringUnchecked(
+        TEST_FILE_CONTENTS, fileUtils.joinPaths(initialRepoFolder, FILE_IN_HEAD));
+    fileInHeadCommitId = repo.commit(repo.getUncommittedFiles(), "Initial commit").getId();
+  }
+
+  private void initAaBase(String initialRepoFolder, String aaBaseFolder) {
+    InitCommand initCommand = component.getInitCommand();
+    InitCommand.basePath.resetValueForTesting();
+    InitCommand.startuposRepo.resetValueForTesting();
+    String[] args = {
+      "--startupos_repo", initialRepoFolder,
+      "--base_path", aaBaseFolder,
+    };
+    initCommand.run(args);
+  }
+
+  private void createAaWorkspace(String name) {
+    WorkspaceCommand workspaceCommand = component.getWorkspaceCommand();
+    String[] args = {"workspace", "-f", name};
+    workspaceCommand.run(args);
+    String repoPath = fileUtils.joinPaths(getWorkspaceFolder(TEST_WORKSPACE), "startup-os");
+    repo = gitRepoFactory.create(repoPath);
+    repo.setFakeUsersData();
+  }
+
+  private void createBlockingStub() throws IOException {
+    // Generate a unique in-process server name.
+    String serverName = InProcessServerBuilder.generateName();
+    server =
+        InProcessServerBuilder.forName(serverName)
+            .directExecutor()
+            .addService(codeReviewService)
+            .build()
+            .start();
+    channel = InProcessChannelBuilder.forName(serverName).directExecutor().build();
+    blockingStub = CodeReviewServiceGrpc.newBlockingStub(channel);
+  }
+
+  public String joinPaths(String first, String... more) {
+    return FileSystems.getDefault().getPath(first, more).toAbsolutePath().toString();
+  }
+
+  private String getWorkspaceFolder(String workspace) {
+    return joinPaths(aaBaseFolder, "ws", workspace);
+  }
+
+  private void writeFile(String contents) {
+    writeFile(TEST_FILE, contents);
+  }
+
+  private void writeFile(String filename, String contents) {
+    fileUtils.writeStringUnchecked(
+        contents, fileUtils.joinPaths(getWorkspaceFolder(TEST_WORKSPACE), "startup-os", filename));
+  }
+
+  private Protos.DiffFilesResponse getExpectedResponse(String commitId) {
+    return Protos.DiffFilesResponse.newBuilder()
+        .addAllBranchInfo(
+            Collections.singleton(
+                com.google.startupos.common.repo.Protos.BranchInfo.newBuilder()
+                    .setDiffId(2)
+                    .setRepoId(REPO_ID)
+                    .addCommit(
+                        com.google.startupos.common.repo.Protos.Commit.newBuilder().setId(commitId))
+                    .build()))
+        .build();
+  }
+
+  private Protos.DiffFilesResponse getResponse(String workspace, long diffNumber) {
+    Protos.DiffFilesRequest request =
+        Protos.DiffFilesRequest.newBuilder().setWorkspace(workspace).setDiffId(diffNumber).build();
+
+    return blockingStub.getDiffFiles(request);
+  }
+
+  @Test
+  public void testFirestoreClientMock() {
```

Comment:
Why do you need this test?

Reply:
It just to check that "mocking" works. I'll delete it.